io/src/main/scala/fs2/io/file/file.scala

```diff
@@ -39,6 +38,21 @@ package object file {
       .flatMap(c => pulls.readRangeFromFileHandle(chunkSize, start, end)(c.resource))
       .stream
 
+  /**
+    * Reads all data synchronously from the file at the specified `java.nio.file.Path`.
+    * Then waits for `delay` duration and reads whatever has been added to the file
+    * in the mean time. Repeats from delay.
+    */
+  def keepReading[F[_]: Sync: ContextShift: Timer](path: Path,
```

Comment:
What happens if the file is deleted by some other process while we're reading?

Reply:
As is, stream does not stop, it just emits nothing more. As if nothing new were being written to the file. It is the same behaviour as `file.readAll` and `file.readRange` if the file is deleted before reading is done.

What is the desired behaviour? For the stream to terminate, for example?

Reply:
Actually, I just looked at `FileHandle`'s implementation, and what I said above is not true.

I will correct the code accordingly.

Reply:
Okay, so... 

`FileHandle`s are based on `java.nio.channels.FileChannel`. `FileChannel`s seem to outlive their underlying files, in the sense that when the file is deleted, it's `FileChannel` can stay open and I couldn't find any way of asking the `FileChannel` wether the file is still there.

So, unless `FileHandle` is essentially modified, there seems to be no way of knowing that the file is deleted once the handle is created.

Any ideas for a solution?